### PR TITLE
Add support for kerberos 01

### PR DESCRIPTION
**Pull request by chifflier**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [X] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: none

Describe changes:
- This PR adds support for a new protocol, Kerberos (RFC 4120)
- It follows exchanges on port 88, both TCP and UDP
- It is written in Rust, and the parser itself is shared with nfsv4 and samba (`kerberos-parser`)
- It adds a logger for some parameters (client and server realm+principal, cryptographic parameters), in JSON
- It adds new rules for events. The rule for weak crypto algorithms is configurable, it can generate tons of alerts

Example of JSON output for an `AS-REP` message (only "krb5" key):
```
  "krb5": {
    "msg_type": "KRB_AS_REP",
    "cname": "testuser",
    "realm": "INTERNET.LOCAL",
    "sname": "krbtgt/INTERNET.LOCAL",
    "encryption": "aes256-cts-hmac-sha1-96"
  }
```

Output for a failed `AS-REP`:
```
  "krb5": {
    "msg_type": "KRB_ERROR",
    "failed_request": "KRB_AS_REQ",
    "error_code": "KDC_ERR_PREAUTH_REQUIRED",
    "cname": "testuser",
    "realm": "INTERNET.LOCAL",
    "sname": "krbtgt/INTERNET.LOCAL",
    "encryption": "<none>"
  }
```

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

### CODE REVIEW

File: rust/src/krb/krb5.rs [hunk 1]
```diff
@@ -0,0 +1,560 @@
...
+    let mut s = String::new();
+    for &b in bytes {
+        s.push_str(&format!("{:02X}", b));
+    }
+    s
+}
+
+impl KRB5State {
+    pub fn new() -> KRB5State {
+        let l = conf_get_bool("app-layer.protocols.krb5.warn-weak-crypto");
```
Comment (victorjulien) [inline rust/src/krb/krb5.rs]: this is done at packet runtime, right? The conf api isn't really designed for that. Can we do this at init somehow?

File: rust/src/krb/log.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * 02110-1301, USA.
+ */
+
+// written by Pierre Chifflier  <chifflier@wzdftpd.net>
+
+use json::*;
+use krb::krb5::{KRB5State,KRB5Transaction};
+
+#[no_mangle]
+pub extern "C" fn rs_krb5_log_json_response(_state: &mut KRB5State, tx: &mut KRB5Transaction) -> *mut JsonT
```
Comment (victorjulien) [inline rust/src/krb/log.rs]: would it make sense to add a field indicating the weak crypto here?

File: rust/src/krb/krb5.rs (hunk 1 above)
Comment (chifflier) [reply rust/src/krb/krb5.rs]: Sure, done in a commit added to the list (ec9f31d). If you prefer me to squash the commits just tell me

**Comment by norg** [comment]
Would you like to add some doc as well :)?

**Comment by chifflier** [comment]
@norg : sure, but the kerberos parser does not any any detection keyword (yet). It does protocol compliance, and raise events in some cases - I could not find any section in the userguide to describe that. Any idea ?

**Comment by norg** [comment]
My bad, then it should be enough to be listed as possible protocol in the intro section which you already did!

**Comment by chifflier** [comment]
Updated by #3371

**Pull request closed without merging**